### ISSUE TITLE
using . (dot) reference syntax on a map fails when value is a function

### ISSUE THREAD

**Issue by vans163** [elixirscript/elixirscript#380]
```elixir
#Good
map = %{test_map: 111}
map.test_map

#Good
map = %{test_map: &Map.get/2}
Map.get(map, :test_map)

#Bad
map = %{test_map: &Map.get/2}
map.test_map

#Uncaught MatchError: No match for: 
```

Note that Map.get/2 still works in the Bad case.

**Comment by bryanjos** [comment]
This may be a harder one to fix correctly. Since Elixir doesn't make a distinction between calling getting a field and calling a function with 0 arity, there is some code to try to handle those cases.

**Comment by vans163** [comment]
Ah I got it, so this happens on elixir side too. Should be closed then I think if its an elixir behavior.

**Comment by bryanjos** [comment]
No I don't think it happens on the Elixir side. Just have to find a better way of handling the above.

**Comment by OvermindDL1** [comment]
Maybe mark module structs in javascript with a special key for the distinction?

**Comment by bryanjos** [comment]
There is the `__struct__` key there already so that helps. And in the case above it's on a map so we should know it's a field being accessed. Actually since it's a map that should be enough information to determine.

**Comment by vans163** [comment]
Im putting in a commit for this but I understand if I am missing a certain case, the code currently does:
```
    if (item_prop instanceof Function || typeof item_prop === 'function') {
      return item.get(prop)();
    }
```
This gets called if we use the . syntax to Acces from a map.  Why is it calling the function there? What case does that take care of?

**Comment by bryanjos** [comment]
@vans163 correct, the code you have removed should be the fix for this.

**Comment by vans163** [comment]
pushed https://github.com/elixirscript/elixirscript/pull/387/commits/aa3b44995435a7382f5e41f8e2ad39841c758f6a

**Comment by bryanjos** [comment]
closed with #387
